base/initdefs.jl

```diff
@@ -53,6 +72,11 @@ function init_bind_addr()
 end
 
 function early_init()
+    @doc """
+        JULIA_HOME
+
+    A string containing the full path to the directory containing the `julia` executable.
+    """ ->
```

Comment:
Would be good to avoid defining docstrings inside functions where possible. Maybe the following would be better?

```julia
"""
    JULIA_HOME

A string containing the full path to the directory containing the `julia` executable.
"""
:JULIA_HOME

function early_init()
    global const JULIA_HOME = ccall(:jl_get_julia_home, Any, ())
    # ...
```

This works since docstrings are associated with `Binding` objects rather than directly with the real object. We only ever check whether it is defined at runtime when retrieving docs.

(Technically you can write `JULIA_HOME` instead of `:JULIA_HOME` and it will also work, but perhaps the quoted form does better at alerting the read of the "lazy binding" taking place.)